server/src/trainee/traineeController.ts

```diff
@@ -0,0 +1,187 @@
+/**
+ * Definition of the handler functions for the members module
+ */
+import bcrypt = require("bcryptjs");
+import { Request, Response } from "express";
+import database = require("../database");
+import { getRandomString } from "../utils/stringUtils";
+import { createMailAccount, addMailAccountToMailingList } from "../utils/plesk";
+import { createMWUser } from "../utils/mediawiki";
+import { createNCUser } from "../utils/nextcloud";
+import * as traineeTypes from "./traineeTypes";
+import { PoolConnection } from "mysql";
+import * as authTypes from "./../global/auth/authTypes";
+import { canPermissionBeDelegated, doesPermissionsInclude } from "../utils/authUtils";
+
+/**
+ * Retrieves choices of mentor, internal project and department of all trainees of given generation
+ * @param req generation_ID
```

Comment:
Please use camel case variables:
```suggestion
 * @param req generationID
```

Reply:
Is the request parameter not more than just the generationID?
The request paramter is an object with a body object which has `generationID` as an attribute.

Reply:
You should also use type annotations for the parameters (see suggestion a few comments below).

Reply:
fixed